woocommerce/woocommerce-gutenberg-products-block
# Update `PaymentMethodIcons`, and `CheckoutButton` components to TypeScript

This PR includes the following changes:
- Create a `PaymentMethodIcon` type.
- Convert files in `assets/js/base/components/cart-checkout/payment-method-icons` to TypeScript
- Convert `CheckoutButton` to TypeScript

### How to test the changes in this Pull Request:

1. Smoke test the Cart/Checkout flow, ensure any payment method icons are displaying correctly. Below the cart summary on the Cart page should also be some payment method icons, ensure these show properly.
2. Go to `assets/js/payment-method-extensions/payment-methods/bacs/index.js` and add `icon="bank"` to the `PaymentMethodLabel` and enable BACS payments in WooCommerce > Settings > Payments.
3. On the Checkout page, the bank transfer payment method should have an icon next to it.
